scripts/examples/gn_to_cmake.py

```diff
@@ -0,0 +1,760 @@
+            if target.cmake_type.is_linkable:
+                libraries.add(cmake_dependency_name)
+            else:
+                nonlibraries.add(cmake_dependency_name)
+
+    # Non-library dependencies.
+    if nonlibraries:
+        out.write('add_dependencies("${target}"')
+        for nonlibrary in nonlibraries:
+            out.write('\n  "')
+            out.write(nonlibrary)
+            out.write('"')
+        out.write(')\n')
+
+    # Non-OBJECT library dependencies.
+    combined_library_lists = [target.properties.get(
+        key, []) for key in ['libs', 'frameworks']]
+    external_libraries = list(itertools.chain(*combined_library_lists))
+    if target.cmake_type.is_linkable and (external_libraries or libraries):
+        library_dirs = target.properties.get('lib_dirs', [])
+        if library_dirs:
+            SetVariableList(
+                out, '${target}__library_directories', library_dirs)
+
+        system_libraries = []
+        for external_library in external_libraries:
+            if '/' in external_library:
+                libraries.add(project.GetAbsolutePath(external_library))
+            else:
+                if external_library.endswith('.framework'):
+                    external_library = external_library[:-len('.framework')]
+                system_library = 'library__' + external_library
+                if library_dirs:
+                    system_library = system_library + '__for_${target}'
+                out.write('find_library("')
+                out.write(CMakeStringEscape(system_library))
+                out.write('" "')
+                out.write(CMakeStringEscape(external_library))
+                out.write('"')
+                if library_dirs:
+                    out.write(' PATHS "')
+                    WriteVariable(out, '${target}__library_directories')
+                    out.write('"')
+                out.write(')\n')
+                system_libraries.append(system_library)
+        out.write('target_link_libraries("${target}"')
+        for library in libraries:
+            out.write('\n  "')
+            out.write(CMakeStringEscape(library))
+            out.write('"')
+        for system_library in system_libraries:
+            WriteVariable(out, system_library, '\n  "')
+            out.write('"')
+        out.write(')\n')
+
+
+def WriteProject(project):
+    out = open(posixpath.join(project.build_path, 'CMakeLists.txt'), 'w+')
+    out.write('# Generated by gn_to_cmake.py.\n')
+    out.write('cmake_minimum_required(VERSION 3.7 FATAL_ERROR)\n')
+    out.write('cmake_policy(VERSION 3.7)\n')
+    out.write('project(MatterAndroid)\n\n')
+
+    out.write('file(WRITE "')
+    out.write(CMakeStringEscape(
+        posixpath.join(project.build_path, "empty.cpp")))
+    out.write('")\n')
+
+    for target_name in project.targets.keys():
+        out.write('\n')
+        WriteTarget(out, Target(target_name, project), project)
+
+    out.close()
+
+
+def main():
```

Comment:
can you add some unit tests for this file? It would be very helpfull visually to see on a specific json file what the expected output is.

Reply:
OK, I will do that, do you think it is OK to add them after this PR?

Reply:
All PRs should have unit tests if feasable.